tools/SourceKit/lib/SwiftLang/SwiftEditorInterfaceGen.cpp

```diff
@@ -155,8 +157,10 @@ class AnnotatingPrinter : public StreamPrinter {
           OS << TargetUSR;
         }
         StringRef USR = OS.str();
-        Info.USRMap[USR] = Entry;
         DeclUSRs.emplace_back(VD, USR);
+        auto iterator = Info.USRMap.insert_or_assign(USR, Entry).first;
+        // Set the USR in the declarations to the key in the USRMap, because the lifetime of that matches/exceeds the lifetime of Decls
+        Info.Decls.back().USR = iterator->getKey();
```

Comment:
Yes, it's not the most elegant solution. The [`StringMap` only copies keys on insertion into the map](https://llvm.org/docs/ProgrammersManual.html#llvm-adt-stringmap-h), and keys are otherwise only invalidated by `erase()`, `clear()` (which just erases all), and destruction. The two functions aren't used on the `USRMap`, but I added an additional comment explaining that, and I annotated the StringMap declaration with a comment requesting to keep the map and the Decls array in sync. Good catch :).